src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -583,15 +591,23 @@ impl EventProcessor {
         self.dnd.result = None;
         if let Ok(mut data) = unsafe { self.dnd.read_data(window) } {
             let parse_result = self.dnd.parse_data(&mut data);
+
             if let Ok(ref path_list) = parse_result {
-                for path in path_list {
-                    let event = Event::WindowEvent {
+                if self.dnd.has_entered {
+                    callback(&self.target, Event::WindowEvent {
                         window_id,
-                        event: WindowEvent::HoveredFile(path.clone()),
-                    };
-                    callback(&self.target, event);
+                        event: WindowEvent::DragMoved { position: self.dnd.position },
```

Comment:
can you return event based on `if` and then just `callback(&self.target, Event::WindowEvent { window_id, event })`?

Reply:
Done.